examples/foomedical/src/pages/MessagesPage.tsx

```diff
@@ -1,122 +1,75 @@
-import { useEffect, useState } from 'react';
+import { useState, useCallback, useMemo } from 'react';
+import { useParams } from 'react-router-dom';
+import { Alert } from '@mantine/core';
+import { Communication, HumanName, Patient, Practitioner } from '@medplum/fhirtypes';
+import { createReference, formatGivenName, getReferenceString, normalizeErrorString } from '@medplum/core';
+import { BaseChat, Document, useMedplum, useMedplumProfile, useResource } from '@medplum/react';
 import { Loading } from '../components/Loading';
+import { showNotification } from '@mantine/notifications';
+import { IconCircleOff } from '@tabler/icons-react';
 
 export function Messages(): JSX.Element {
   const medplum = useMedplum();
   const profile = useMedplumProfile() as Patient;
-  const [messages, setMessages] = useState<Communication[]>();
+  const profileRef = useMemo(() => (profile ? createReference(profile) : undefined), [profile]);
+  const [communications, setCommunications] = useState<Communication[]>([]);
+  const { practitionerId } = useParams();
+  const practitioner = useResource<Practitioner>({
+    reference: `Practitioner/${practitionerId}`,
+  });
 
-  useEffect(() => {
-    medplum
-      .graphql(
-        `
-        {
-          CommunicationList(subject: "${getReferenceString(profile)}") {
-            resourceType
-            id
-            meta {
-              lastUpdated
-            }
-            payload {
-              contentString
-              contentAttachment {
-                url
-                contentType
-              }
-            }
-            sender {
-              reference
-              resource {
-                ... on Patient {
-                  resourceType
-                  id
-                  name {
-                    given
-                    family
-                  }
-                  photo {
-                    contentType
-                    url
-                  }
-                }
-                ... on Practitioner {
-                  resourceType
-                  id
-                  name {
-                    given
-                    family
-                  }
-                  photo {
-                    contentType
-                    url
-                  }
-                }
-              }
-            }
-          }
+  const sendMessage = useCallback(
+    (content: string): void => {
+      if (!practitioner) {
+        return;
       }
-        `
-      )
-      .then((value) => setMessages(value.data.CommunicationList as Communication[]))
-      .catch((err) => console.error(err));
-  }, [medplum, profile]);
 
-  if (!messages) {
+      if (!profileRef) {
+        return;
+      }
+
+      const practitionerRef = createReference(practitioner);
+
+      medplum
+        .createResource<Communication>({
+          resourceType: 'Communication',
+          status: 'in-progress',
```

Comment:
I noticed the old code used the `status: 'completed'` so I wonder about the semantics, whether we use `in-progress` to indicate that the conversation is open or `completed` to indicate that the message has been sent. Perhaps Rahul can confirm that.

Reply:
Great question. We actually document the lifecyle here: https://www.medplum.com/docs/communications/organizing-communications#representing-individual-messages

As long as the `sent` is populated, we will consider this in transit. 

This brings up a good point: 
@filipeximenes , to mark the message as received, you should use the `onMessageReceived` prop to do 2 things: 
  - set the received time
  - mark as completed